interface.go

```diff
@@ -27,6 +27,23 @@ type Token struct {
 	SentTime     time.Time
 }
 
+// A ClientToken is a token received by the client.
+// It can be used to skip address validation on future connection attempts.
+type ClientToken struct {
+	data []byte
+}
+
+type TokenCache interface {
```

Comment:
Be explicit that this is conceptually a multimap.

I'm also not happy about the name cache – this isn't caching, it's storing (and deleting).

Reply:
I'm not sure I understand what you mean here.